plugin-template

```diff
@@ -37,6 +37,7 @@ DEFAULT_SETTINGS = {
     "docker_fixtures": False,
     "docs_test": True,
     "issue_tracker": "github",
+    "parallel_test_workers": 0,
```

Comment:
I just started to ask myself, whether it is worth making this an option. Current tests will not be accidentally marked as parallel.
The value i see in this option is if you intend to make parallel the default and want plugins do declare tests a serial.

Reply:
Oh, i just realized, it's not-a-boolean for a reason.

Reply:
I like the idea that the number of parallelization is configurable, but I do think it would be better if we could make the default 8 instead of zero. The one blocking reason is we need every plugin to receive `pytest-xdist`. Right now I have pulpcore enabling that [here](https://github.com/pulp/pulpcore/pull/1810/files#diff-93394e9ecb9acc4a48dffa8515bf0d6c45d2170dd7fd2dcc76d1bd722d94fa0fR8) to do that. If we wanted to make this 8 we could have pulp-smash declare `pytest-xdist` as a dependency.... I think I should do that, merge it, and then change this default to 8. What do you think about this proposal?

Reply:
Sounds like a plan. My underlying question is still, in the long run do you expect we mark tests as viable for parallel or mark them for needsbe serial?

Reply:
In the very long run I expect the mark to be serial. To do that though I'd have to have all tests auto-mark as `serial` and to do that I think projects would have to take an action that I can't provide automatically to them in the plugin template. So for that reason I think leaving it as parallel for now is the best I can do. Wdyt? Suggestions are welcome.

Reply:
Also this is ready: https://github.com/pulp/pulp-smash/pull/1266